From 50dedcb19ce7dad6fa10a2ab25ed5c42badaf8cf Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Fri, 21 Jun 2019 20:21:52 +0200
Subject: [PATCH 37/37] SystemUI: Disallow changing disabled signal icon's
 visibility state

* Fixes an issue where disabled SIM icon gets showed up
  after toggling airplane mode on and off when other SIM
  has mobile data turned off.

Change-Id: I546a326eb52b0496f5fbd9057d368b1a90827d58
---
 .../com/android/systemui/statusbar/StatusBarMobileView.java   | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarMobileView.java b/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarMobileView.java
index 45bf8268d2b..dbffafcad45 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarMobileView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarMobileView.java
@@ -245,6 +245,10 @@ public class StatusBarMobileView extends FrameLayout implements DarkReceiver,
             return;
         }
 
+        if (mState != null && (!mState.visible || !mState.provisioned)) {
+            return;
+        }
+
         mVisibleState = state;
         switch (state) {
             case STATE_ICON:
-- 
2.17.1

